### PR TITLE
Feat/edit nav bar page

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR adds in an edit nav bar page ~(currently accessible through sites/{sitename}/menu)~ accessible through workspace. To be reviewed in conjunction with PR#[122](https://github.com/isomerpages/isomercms-backend/pull/112) on the isomercms-backend repo.

Miscellaneous items to clean up:
- ~Proptypes for nav and sublink sections~
- ~Link to editNav from workspace~
- Error handling?

Points for discussion:
- Currently the display for the nav bar doesn't exactly match the one on the actual site - it is missing the logo, and the nav bar on the cms is scrollable while the actual one is not. I don't see a good way for the two to match, since half the page on the cms is dedicated to the editable portion
- Errors - what kind of errors (if any) should we look out for? Currently no error checking has been implemented since only 2 fields can take in any user input (title and url), and these can take on any values if editing the file through github. Should we enforce something like maximum/minimum length?

### CODE REVIEW

**Review by kwajiehao — COMMENTED** [review]
### Missing external link functionality
The main thing that's missing from this PR is that we can't define an external link (the external link should also have a link out icon in the template - see below)

Screenshot 2021-02-23 at 1 39 21 AM

### Bug in displaying collections

The dropdown in the nav bar does not display the collection pages / subfolders.

Screenshot 2021-02-23 at 1 46 45 AM

How we proceed for this part depends on whether we are going to release the `EditNavBar` first or if we are going to release it at the same time as the broader navigation changes we're making. If we're releasing it together with the broader navigation changes, it might make sense to retrieve the collection data from the directory config file.

### Summary
Other than that, it looks good to me overall. The code was clearly consistent with `EditHomepage` and `EditContactUs` which made it easy to follow.

Re: points of discussion:
- for the appearance of the `EditNavBar`, let's push this to staging and get feedback from the designers
- for error-handling, I agree that there is little that we can enforce. I think we can perform the same validation that we do for the same `title` and `url`/`permalink` fields in `EditHomepage`. We can also get more feedback from the designers once it's in staging

couple of comments:
- left a comment on reducing code repetition for the template
- can we modify the behavior of `Create new link` so that if a resource room is already defined, a user cannot add a new resource room to the nav bar? This behavior is already implemented in `EditHomepage`

File: src/templates/NavBar.jsx
Comment (alexanderleegs) [reply src/templates/NavBar.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/dd3ceaba3b7e7db9dc56cd22976382e89cfa11c2

File: src/utils.js
Comment (alexanderleegs) [reply src/utils.js:49]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/dd3ceaba3b7e7db9dc56cd22976382e89cfa11c2

File: src/components/navbar/NavSection.jsx
Comment (alexanderleegs) [reply src/components/navbar/NavSection.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/dd3ceaba3b7e7db9dc56cd22976382e89cfa11c2

File: src/templates/NavBar.jsx
Comment (alexanderleegs) [reply src/templates/NavBar.jsx]: Good point, resolved in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/d0a2195bee3ae521fac4f5f452fee5dca0518e88

**Comment by alexanderleegs** [comment]
> ### Missing external link functionality
> The main thing that's missing from this PR is that we can't define an external link (the external link should also have a link out icon in the template - see below)

As discussed, we will resolve this in a future PR, along with the other references to external links. https://github.com/isomerpages/isomercms-frontend/issues/351
 
> ### Bug in displaying collections
> The dropdown in the nav bar does not display the collection pages / subfolders.

As discussed, we will hold off on this until the new folder endpoints have been merged. https://github.com/isomerpages/isomercms-frontend/issues/352


> * for error-handling, I agree that there is little that we can enforce. I think we can perform the same validation that we do for the same `title` and `url`/`permalink` fields in `EditHomepage`. We can also get more feedback from the designers once it's in staging

Error handling added in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/d3e1e060be523faa7f2901c002fdb01973d4693f

> * can we modify the behavior of `Create new link` so that if a resource room is already defined, a user cannot add a new resource room to the nav bar? This behavior is already implemented in `EditHomepage`

Good catch, resolved in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/398873debc0fe4650514dc686c06b991933559d3

**Review by kwajiehao — CHANGES_REQUESTED** [review]
"Create new link" button is unable to create a new collection link - it ends up creating a new single page link instead

File: src/layouts/EditNavBar.jsx
Comment (alexanderleegs) [reply src/layouts/EditNavBar.jsx]: Renamed all state variables in the section on mount in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/52000cfd42ac4a3b96d5fc9f708cc209db504579

**Comment by alexanderleegs** [comment]
> "Create new link" button is unable to create a new collection link - it ends up creating a new single page link instead

Good catch, I've disabled collection links for repos which have no collections in https://github.com/isomerpages/isomercms-frontend/pull/348/commits/812692e905c40b730857208117c47cb997b1eda6

**Review by kwajiehao — APPROVED** [review]
lgtm

**Pull request merged**
